fastlane/lib/fastlane/fast_file.rb

```diff
@@ -235,26 +236,72 @@ def import_from_git(url: nil, branch: 'HEAD', path: 'fastlane/Fastfile')
 
         # Checkout the repo
         repo_name = url.split("/").last
+        checkout_param = branch
 
         Dir.mktmpdir("fl_clone") do |tmp_path|
           clone_folder = File.join(tmp_path, repo_name)
 
-          branch_option = ""
           branch_option = "--branch #{branch}" if branch != 'HEAD'
 
-          clone_command = "GIT_TERMINAL_PROMPT=0 git clone '#{url}' '#{clone_folder}' --depth 1 -n #{branch_option}"
-
           UI.message "Cloning remote git repo..."
-          Actions.sh(clone_command)
+          Actions.sh("GIT_TERMINAL_PROMPT=0 git clone '#{url}' '#{clone_folder}' --depth 1 -n #{branch_option}")
+
+          unless version.nil?
+            UI.message "Fetching remote git tags..."
+            Actions.sh("cd '#{clone_folder}' && GIT_TERMINAL_PROMPT=0 git fetch --all --tags -q")
+
+            # Separate version from optimistic operator
+            split_version = version.split(" ")
+            version_number = split_version.last
+            operator = split_version.first
+
+            # Fetch all possible tags
+            git_tags_string = Actions.sh("cd '#{clone_folder}' && git tag -l")
+            git_tags = git_tags_string.split("\n")
+
+            # Delete tags that are not a real version number
+            git_tags.delete_if { |tag| Gem::Version.correct?(tag) != 0 }
+
+            # Sort tags based on their version number
+            git_tags.sort_by { |tag| Gem::Version.new(tag) }
+
+            # ~> should select the latest version withing constraints.
+            # -> should select a specific version without fallback.
+            if operator == "~>"
+              # Drop last specified digit in version
+              last_dot_index = version_number.rindex('.')
+              version_range = version_number[0..last_dot_index - 1]
+
+              # Search matching version in array
+              matching_git_tags = git_tags.select do |tag|
+                tag.start_with?(version_range)
+              end
+
+              UI.user_error!("No version found within the \"#{version_range}.*\" range") if matching_git_tags.count == 0
+              UI.user_error!("Specified version \"#{version_number}\" is to high. Latest version within this range is: \"#{matching_git_tags.last}\"") if Gem::Version.new(matching_git_tags.last) < Gem::Version.new(version_number)
+              checkout_param = matching_git_tags.last
+
+            elsif operator == "->" || split_version.count == 1
+              # Search matching version in array
+              matching_git_tags = git_tags.select do |tag|
+                tag == version_number
+              end
+
+              UI.user_error!("The specified version \"#{version_number}\" doesn't exist") if matching_git_tags.count == 0
+              checkout_param = matching_git_tags.last
+            else
+              UI.user_error!("The specified operator \"#{operator}\" in \"#{version}\" is unknown. Please use one of these '~> ->'")
+            end
+          end
```

Comment:
Any chance we could pull this into it's own method? Might be nice to just not have so much nested code right here in line.